public/scripts/world-info.js

```diff
@@ -3269,12 +3278,13 @@ async function _save(name, data) {
     eventSource.emit(event_types.WORLDINFO_UPDATED, name, data);
 }
 
-async function saveWorldInfo(name, data, immediately) {
+async function saveWorldInfo(name, data, immediately = false) {
     if (!name || !data) {
         return;
     }
 
-    worldInfoCache.delete(name);
+    // Update cache immediately, so any future call can pull from this
+    worldInfoCache.set(name, structuredClone(data));
```

Comment:
You don't have to clone here since it's done in the util class

Reply:
Totally correct. That happens if you are sleepy, and later build a util class. Thank you.